### PR TITLE
fix(dialog): Fix dialog corners

### PR DESCRIPTION
**Pull request by trezy**
**Description**
The dialog styles added by @sophi-est weren't being loaded into the compiled stylesheet. I've added
them to the elements index file and moved the rounded corners styles into and `.is-rounded` modifier
class.

**Compatibility**
This restores functionality that was lost, does not remove or break other functionality.

**Caveats**
The rounded corners for `<dialog />` components necessarily *are not* the same as `.container` components. This is because `<dialog />` elements are intended to be displayed over top of other elements and, with the way we create pixelated corners currently, the `<dialog />` component can't receive a background color.

**Testing Instructions**
1. Load Storybook (`npm run storybook`)
1. Verify that `<dialog />` components still work
1. Verify that the `<dialog />` component's `.is-rounded` state modifier works as expected.
1. ???
1. Profit

### CODE REVIEW

File: scss/elements/dialogs.scss [hunk 1]
```diff
@@ -12,4 +9,10 @@
   > :last-child {
     margin-bottom: 0;
   }
+
+  &.is-rounded {
+    color: $base-color;
+    border: none;
+    box-shadow: 4px 0, -4px 0, 0 4px, 0 -4px;
```
Comment (jjspace) [inline scss/elements/dialogs.scss]: Are we sure that we always want the border to match the color of the text? I could easily see someone wanting different colored text within a dialog while maintaining a black border. With the "default color" solution of box shadows this is not possible. That's why I did not make this change when updating most elements to use the default color mentioned in issue #60

Comment (trezy) [reply scss/elements/dialogs.scss]: @jjspace: Yeah, I agree with you. However, this is the way that we handle it in a few different spots in the framework. I think it’s fine for now, but we should absolutely investigate alternatives to the box-shadow method we’re currently using for borders.

Comment (jjspace) [reply scss/elements/dialogs.scss]: Right, I understand that but I checked all the places that `box-shadow` is used without colors specified and all but one instance are changes that I made in PR #143 where I took special care not to change it in places that could potentially have text in them for this very reason. The only other spot that uses the `color` property for the `box-shadow` is in `.nes-progress` class which will never have text inside so it shouldn't be an issue.
IMHO if the aim is to stick to what we currently do I think the colors should be defined in the `box-shadow` and not rely on `color` for the `<dialog />` component which will almost always have text inside it.

Comment (trezy) [reply scss/elements/dialogs.scss]: Yeah @jjspace, I understand what you're saying now. I've fixed the issue. 😉

**Pull request merged**